### ISSUE TITLE
Dark Mode

### ISSUE THREAD

**Issue by helloanoop** [usebruno/bruno#23]
Bruno's layout is inspired by VSCode layout and colors.
For the dark theme colors, please refer the vscode dark theme.

Goal here is to get a dark theme out soon (based on vscode dark theme)

PS: In the future, we will add support for custom themes.

**Comment by helloitsm3** [comment]
I can contribute to this

**Comment by helloanoop** [comment]
I am late @helloitsm3 Just saw your PR now.. lol..
Thanks! I will review this in the next couple of hours.

**Comment by helloanoop** [comment]
Hi @helloitsm3 

Thank you so much for taking the time to contribute and make bruno better.

Here are some comments.
### Prettier
This makes sense, thank you for setting up the config and making the changes. I feel this will help us maintain a consistent standard as more people contribute.

I just have one per peeve (if you don't mind), lets keep the semicolons.
I know it's not required in javascript (except for rare occassions), but I'd rather have it instead of worrying about what that rare occasion is. This preference may also be because of me coding in JS since 2012, and during those days it was considered to have the semi-colons as the browser compatibility was not great as it is today.

### Theming
I went through the pattern of how you have implemented themeing, and I feel sad to ask you to consider a change given the amount of time you have spent on the scss config.

I would like to use the `<ThemeProvider theme={theme} />` pattern from `styled-components`

Here is my reasoning.
1. With CSS, we are restricted to flat structure
```css
- color-component-a: color-a;
- color-component-b: color-b;
- color-component-c: color-c;
```
With Theme provider, we can get the benefit of a nested config, which is more expressive
```js
expor const dark = {
  sidebar: {
    background: ""
  }
}
```

You can refer the implementation here https://github.com/helloanoop/kgraph/blob/main/renderer/themes/light.js (one of my unfinished projects)

This will also give us help us to implement custom themes in the future easily. The custom theme can extend the base theme(light/dark) which is easier to do if the format is json.

Let me know if you see the benefits of still going through the scss based route. I'd love to hear if you have a different take on the subject.
   
### Local storage
Thanks for this. Makes sense to store the chosen theme string in local storage.
Love the way you implemented the hook.
I would like you to make one change. Instead of the key being `THEME`, lets store it as `bruno.theme`

**Comment by helloanoop** [comment]
Hey @helloitsm3 
Thanks for factoring in the changes.

The diff is too huge (220 files), I know it's due to prettier.
But it's getting pretty hard to locate the dark mode changes related to the PR.

I have added prettier on the main branch.
Also, I'd like prettier to be only on the react(packages/bruno) packages and not on the node packages.

Please sync the pull main to your fork and create a branch with dark mode changes.

Apologies again to have you copy paste the dark mode work to a new branch and reraise a PR.
It's taxing to go through such a huge diff.
I hope you understand.

**Comment by helloitsm3** [comment]
> 

Ya no worries. I'll make another PR for this. Thanks

**Comment by helloitsm3** [comment]
Noticed that the dark mode wasn't added to the main branch. Was there an issue with the PR? @helloanoop

**Comment by helloanoop** [comment]
@helloitsm3 I just merged it and now is deployed to the play.bruno.com

Thanks so much for your work!

**Comment by helloitsm3** [comment]
> @helloitsm3 I just merged it and now is deployed to https://play.usebruno.com
> 
> Thanks so much for your work!

Great. I'll be working on more themes

**Comment by helloanoop** [comment]
@helloitsm3 @anusreesubash  Release v0.2.0 with your work on dark mode is out. Thanks again for the work on this.
You can download the latest versions at https://www.usebruno.com/downloads

In case you love dark more, its included in the v0.2.0 release
Currently its based on vscode dark theme, but plan to support custom themes in the near future.

PS: @helloitsm3 Let's discuss the custom themes in the new issue you just filed. Closing this now.
